src/main/java/com/contrast/labs/ai/mcp/contrast/AssessService.java

```diff
@@ -217,89 +200,71 @@ public List<VulnLight> listVulnsByAppId(
 
 
 
-    @Tool(name = "list_vulnerabilities_by_application_and_session_metadata", description = "Takes an application name ( app_name ) and session metadata in the form of name / value. and returns a list of vulnerabilities matching that application name and session metadata.")
-    public List<VulnLight> listVulnsInAppByNameAndSessionMetadata(
-            @ToolParam(description = "Application name") String app_name,
+    @Tool(name = "list_vulns_by_app_and_metadata", description = "Takes an application ID (appID) and session metadata in the form of name / value. and returns a list of vulnerabilities matching that application ID and session metadata. Use list_applications_with_name first to get the application ID from a name.")
+    public List<VulnLight> listVulnsByAppIdAndSessionMetadata(
+            @ToolParam(description = "Application ID") String appID,
             @ToolParam(description = "Session metadata field name") String session_Metadata_Name,
             @ToolParam(description = "Session metadata field value") String session_Metadata_Value) throws IOException {
-        logger.info("Listing vulnerabilities for application: {}", app_name);
-        ContrastSDK contrastSDK = SDKHelper.getSDK(hostName, apiKey, serviceKey, userName,httpProxyHost, httpProxyPort);
+        logger.info("Listing vulnerabilities for application: {}", appID);
 
         logger.info("metadata : " + session_Metadata_Name+session_Metadata_Value);
 
-        logger.debug("Searching for application ID matching name: {}", app_name);
-
-        Optional<Application> application = SDKHelper.getApplicationByName(app_name, orgID, contrastSDK);
-        if(application.isPresent()) {
-            try {
-                List<VulnLight> vulns =  listVulnsByAppId(application.get().getAppId());
-                List<VulnLight> returnVulns = new ArrayList<>();
-                for(VulnLight vuln : vulns) {
-                    if(vuln.sessionMetadata()!=null) {
-                        for(SessionMetadata sm : vuln.sessionMetadata()) {
-                            for(MetadataItem metadataItem : sm.getMetadata()) {
-                                if(metadataItem.getDisplayLabel().equalsIgnoreCase(session_Metadata_Name) &&
-                                        metadataItem.getValue().equalsIgnoreCase(session_Metadata_Value)) {
-                                    returnVulns.add(vuln);
-                                    logger.debug("Found matching vulnerability with ID: {}", vuln.vulnID());
-                                    break;
-                                }
+        try {
+            List<VulnLight> vulns =  listVulnsByAppId(appID);
+            List<VulnLight> returnVulns = new ArrayList<>();
+            for(VulnLight vuln : vulns) {
+                if(vuln.sessionMetadata()!=null) {
```

Comment:
to reduce some of the indentation in the this and following statements, I'd invert this and following if-statement checks.